### PR TITLE
chore: Add extension dependency redhat.vscode-commons

### PR DESCRIPTION
**Pull request by mathur07**

### CODE REVIEW

**Comment by fbricon** [comment]
LGTM

**Comment by mathur07** [comment]
[test]

File: test/extension.test.ts [hunk 1]
```diff
@@ -8,38 +8,36 @@ suite('Fabric8 Analytics Extension', () => {
     assert.ok(vscode.extensions.getExtension('redhat.fabric8-analytics'));
   });
 
-  test('should activate', function () {
+  test('should activate', async function () {
```
Comment (arajkumar) [inline test/extension.test.ts:11]:
```suggestion
  test('should activate', async () => {
```

Anyways you are adding async, change it to arrow functions

File: test/extension.test.ts [hunk 2]
```diff
@@ -8,38 +8,36 @@ suite('Fabric8 Analytics Extension', () => {
...
+      return (
+        FABRIC8_COMMANDS.indexOf(value) >= 0 ||
+        value.startsWith('extension.fabric8')
       );
     });
+    assert.equal(
+      foundFabric8Commands.length,
+      FABRIC8_COMMANDS.length,
+      'Some fabric8 commands are not registered properly or a new command is not added to the test'
+    );
```
Comment (arajkumar) [inline test/extension.test.ts:40]: Or use equivalent function from sinon 
```suggestion
      assert.ok(FABRIC8_COMMANDS.includes(...commands);
```

File: .vscode/launch.json [hunk 3]
```diff
@@ -10,21 +10,31 @@
...
+			],
 			"preLaunchTask": "npm: webpack-dev"
 		},
 		{
 			"name": "Launch Tests",
 			"type": "extensionHost",
 			"request": "launch",
 			"runtimeExecutable": "${execPath}",
-			"args": ["${workspaceFolder}/test/resources/sampleNodeApp/", "--disable-extensions", "--extensionDevelopmentPath=${workspaceRoot}", "--extensionTestsPath=${workspaceRoot}/out/test" ],
+			"args": [
```
Comment (arajkumar) [inline .vscode/launch.json:28]: Why --disable-extensions is removed? Anyways our testing should independent of vscode-commons. Its absence shouldn't make test failures.

File: package.json [hunk 4]
```diff
@@ -39,7 +42,7 @@
   ],
   "icon": "icon/openshift_logo.png",
   "engines": {
-    "vscode": "^1.49.0"
+    "vscode": "^1.50.0"
```
Comment (arajkumar) [inline package.json:45]: What is the reason for bumping the version?

File: test/runTest.ts [hunk 5]
```diff
@@ -1,19 +1,32 @@
...
         const extensionDevelopmentPath = path.resolve(__dirname, '../../');
-
         // The path to the extension test script
         // Passed to --extensionTestsPath
         const extensionTestsPath = path.resolve(__dirname, './');
-
+        const vscodeExecutablePath = await downloadAndUnzipVSCode('1.50.0');
+        const cliPath = resolveCliPathFromVSCodeExecutablePath(vscodeExecutablePath)
+        // Use cp.spawn / cp.exec for custom setup
+        cp.spawnSync(cliPath, ['--install-extension', 'redhat.vscode-commons'], {
```
Comment (arajkumar) [inline test/runTest.ts:20]: We shouldn't do this.

File: test/vscontext.mock.ts [hunk 6]
```diff
@@ -10,6 +10,9 @@ class DummyMemento implements vscode.Memento {
     dummyMomentoData[key] = value;
     return Promise.resolve(dummyMomentoData);
   }
+  public setKeysForSync(keys: string[]): void {
```
Comment (arajkumar) [inline test/vscontext.mock.ts:13]: Is this the side-effect of bumping engine to ^1.50.0

**Review by arajkumar — CHANGES_REQUESTED** [review]

File: test/vscontext.mock.ts (hunk 6 above)
Comment (mathur07) [reply test/vscontext.mock.ts:13]: Yes, this has been added to globalState ^1.50.0

Comment (mathur07) [reply test/vscontext.mock.ts:13]: to get it in sync with https://github.com/redhat-developer/vscode-commons/blob/master/package.json#L17

File: package.json (hunk 4 above)
Comment (mathur07) [reply package.json:45]: to get it in sync with https://github.com/redhat-developer/vscode-commons/blob/master/package.json#L17

File: .vscode/launch.json (hunk 3 above)
Comment (mathur07) [reply .vscode/launch.json:28]: as we have added in the extentionDependency test case try to check if vscode-commons is present.

File: test/extension.test.ts (hunk 2 above)
Comment (mathur07) [reply test/extension.test.ts:40]: ack

File: test/extension.test.ts (hunk 1 above)
Comment (mathur07) [reply test/extension.test.ts:11]: ack

File: test/runTest.ts (hunk 5 above)
Comment (fbricon) [reply test/runTest.ts:20]: Seems legit to me : https://github.com/microsoft/vscode-java-test/blob/master/test/index.ts#L17

Comment (arajkumar) [reply test/runTest.ts:20]: > Seems legit to me : https://github.com/microsoft/vscode-java-test/blob/master/test/index.ts#L17
@fbricon, to be specific, my worry is that we shouldn't endup having hard dependency on vscode-commons. How to ensure the extension still works without having a dependency on vscode-commons.?

Comment (fbricon) [reply test/runTest.ts:20]: We need a hard dependency on vscode-commons, else it'll never be installed automatically and you'll never get telemetry

Comment (arajkumar) [reply test/runTest.ts:20]: Ack!

**Review by arajkumar — APPROVED** [review]
lgtm

**Pull request merged**